src/main/java/core/CoreConstants.java

```diff
@@ -34,4 +34,15 @@
     public enum GameEvents {
         GAME_OVER, ROUND_OVER, TURN_OVER, ACTION_CHOSEN
     }
+
+    /**
+     * Used in Components that contain other Components (see IComponentContainer) to mark which players can see the
+     * contents.
+     * MIXED_VISIBILITY is an indicator that none of the previous three apply, and that the IComponentContainer
+     * will need to implement more sophisticated logic. This is done for example in PartialObservableDeck - and
+     * this should cover almost all future eventualities.
+     */
+    public enum VisibilityMode {
+        VISIBLE_TO_ALL, HIDDEN_TO_ALL, VISIBLE_TO_OWNER, MIXED_VISIBILITY
```

Comment:
Some others that I'd expect to see often would be FIRST_VISIBLE (to all) or LAST_VISIBLE (to all), for the cases where you have a deck on the table that's technically face up, so you can see the first card, but you can't look through it, so the others are technically hidden. Then if a card is removed from the deck the next one on the deck would become the one visible. An immediate example of use would be in Colt Express for the round cards where visibility of the first card is set manually, but I imagine more could use this as a default implemented functionality.

Reply:
Nice idea!